CRM/Core/BAO/UFField.php

```diff
@@ -207,7 +207,13 @@ public static function duplicateField($params) {
     $ufField->field_type = $params['field_type'];
     $ufField->field_name = $params['field_name'];
     $ufField->website_type_id = CRM_Utils_Array::value('website_type_id', $params);
-    $ufField->location_type_id = CRM_Utils_Array::value('location_type_id', $params);
+    if (is_null(CRM_Utils_Array::value('location_type_id', $params, ''))) {
```

Comment:
This statement doesn't make sense to me. Testing `is_null(CRM_Utils_Array::value('bar', $foo, ''))` will always return false because:

- If `$foo['bar']` is not set or is null, the default value (empty string) will be used and the condition will be false.
- If `$foo['bar']` is any other value the condition will be false.

Reply:
Correction. Ah I see, I was wrong about that because it's testing `array_key_exists` not `isset` in `CRM_Utils_Array::value`